### PR TITLE
fix gas filters removal

### LINKED ISSUE #61045: Gas mask filtering internals
Reporting client version: 514.1561

## Round ID:
[168483](https://scrubby.melonmesa.com/round/168483)

## Testmerges:
- [Basic Mobs: the cooler simple mobs that run on datum AI. (With reworked cockroach AI as proof of concept)](https://www.github.com/tgstation/tgstation/pull/60694)

## Reproduction:

>Play on TG
>Recieve PR's from ghilker
>Be an average plasmeme
>Wanna use cool gas mask instead of cringe default breath mask
>Gas mask filter filtering your internals so you gasp for air
>You can't remove filter(tried normal human interaction)

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Fix issue with removing the filter, now it properly work
fix #61045 

## Why It's Good For The Game
Issue fix 

## Changelog
:cl:
fix: rclick on gasmasks removes filters
/:cl:

### CODE REVIEW

File: code/modules/clothing/masks/gasmask.dm [hunk 1]
```diff
@@ -37,6 +37,7 @@
 		. += "<span class='notice'>[src] has [max_filters] slot\s for filters.</span>"
 	if(LAZYLEN(gas_filters) > 0)
 		. += "<span class='notice'>Currently there [LAZYLEN(gas_filters) == 1 ? "is" : "are"] [LAZYLEN(gas_filters)] filter\s with [get_filter_durability()]% durability.</span>"
+		. += "<span class='notice'>The filters can be removed by right-clicking with an empty hand on [src].</span>"
```
Comment (LemonInTheDark) [inline code/modules/clothing/masks/gasmask.dm:40]: The text says right clicking with an empty hand. Isn't `attack_self_secondary` from being in someone's active hand, and right clicked? Rather then right clicking with an empty hand, which is maybe a bit more intuitive

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
What was the actual error behind it?

File: code/modules/clothing/masks/gasmask.dm (hunk 1 above)
Comment (Ghilker) [reply code/modules/clothing/masks/gasmask.dm:40]: Well it works if you click it with an empty hand too, but if you want I'll change it

**Comment by Ghilker** [comment]
> What was the actual error behind it?

Something changed about attackby_secondary but no idea what

**Comment by LemonInTheDark** [comment]
Did some testing on a local. Looks like attackby_secondary is meant for only being right clicked by someone already holding an item. 
image
I think what you want here is `attack_hand_secondary`, which is someone right clicking something with an empty hand. Since you're guaranteed they have a free hand, you can have it transfer the filter into their hand too, which is cool and nice. (If you do this return `SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN` so the code doesn't runtime)

**Pull request merged**